<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Document</title>
</head>

<body>
    <script src="../jquery.js"></script>
    <script>
        // promise的封装
        function request(url) {
            return new Promise((resolve, reject) => {
                $.ajax({
                    url: url,
                    success(data) {
                        resolve(data);
                    },
                    error(err) {
                        reject(err);
                    },
                });
            });
        }
        //   world的请求 promise对象
        request("http://localhost:3000/data")
            //   then中的第一个函数是 成功的回调  第二个函数是失败的回调
            .then(
                (res) => {
                    console.log(res);
                },
                (err) => {
                    console.log(err);
                }
            );
        //等价于上面
        request("http://localhost:3000/data")
            .then((res) => {
                console.log(res);
            })
            .catch((err) => {
                console.log(err);
            });
    </script>
</body>

</html>